## Hypothesis Testing

In [ ]:
import numpy as np
import pandas as pd
from scipy.stats import chi2_contingency
from scipy.stats import ttest_ind

df = pd.read_csv('C:/Users/win10/Desktop/Project_Oct25/impactsenseAI-Infosys-Intern-project/data/preprocessed_earthquake_data.csv')

#### 1. Magnitude Difference by Earthquake Type
Hypothesis: The mean earthquake magnitude differs significantly between reviewed and non-reviewed events.

In [ ]:
# Select relevant groups, drop missing values
group1 = df[df['status'] == 'reviewed']['mag'].dropna()
group2 = df[df['status'] != 'reviewed']['mag'].dropna()

ttest_result = ttest_ind(group1, group2, equal_var=False)
print('t-test statistic:', ttest_result.statistic)
print('p-value:', ttest_result.pvalue)

t-test statistic: -1.0748565111189015
p-value: 0.2849496659726232


**Statement**:The t-test result for Hypothesis-1 yields a t-statistic of -1.07 and a p-value of 0.285. Since the p-value is greater than the common significance threshold of 0.05, there is not enough evidence to reject the null hypothesis. Thus, the mean earthquake magnitude does not differ significantly between reviewed and non-reviewed events in your data.

#### 2. Depth Difference by Magnitude Type
Hypothesis: The mean depth of earthquakes significantly varies between 'mb' and 'mw' magnitude type categories.

In [ ]:
group1 = df[df['magType'] == 'mb']['depth'].dropna()
group2 = df[df['magType'] == 'mw']['depth'].dropna()

ttest_result = ttest_ind(group1, group2, equal_var=False)
print('t-test statistic:', ttest_result.statistic)
print('p-value:', ttest_result.pvalue)

t-test statistic: 22.514758657461467
p-value: 8.933021714439036e-112


**Statement**: The t-test result for this hypothesis yields a t-test statistic of 22.51 and a p-value of approximately $8.93 \times 10^{-112}$. Since this p-value is extremely small and far below the conventional significance threshold of 0.05, there is extremely strong evidence to reject the null hypothesis. Therefore, the mean depth of earthquakes is significantly different between `mb` and `mw` magnitude type events in your dataset.

#### 3. Magnitude Consistency Across Months
Hypothesis: There is no significant association between the month of occurrence and the categorical bins of earthquake magnitude (such as low, medium, high).

In [ ]:
# Bin magnitudes into categories
df['mag_bin'] = pd.cut(df['mag'], bins=[0, 3.5, 5.0, 6.5, 9], labels=['Low', 'Medium', 'High', 'Very High'])

# Contingency table: mag_bin vs Month
contingency = pd.crosstab(df['mag_bin'], df['Month'])
chi2, p, dof, expected = chi2_contingency(contingency)
print('Chi-square statistic:', chi2)
print('Degrees of freedom:', dof)
print('p-value:', p)

Chi-square statistic: 46.542726462177434
Degrees of freedom: 33
p-value: 0.05922370884750974


**Statement**: The chi-square test result for this hypothesis yields a chi-square statistic of 46.54, degrees of freedom of 33, and a p-value of approximately 0.0592. Since this p-value is slightly above the conventional significance threshold of 0.05, there is insufficient evidence to reject the null hypothesis at the 5% level. Therefore, the data does not provide strong enough evidence to conclude that earthquake magnitude category and month of occurrence are dependent variables.

#### 4. DayOfWeek Consistency Across Event Type
Hypothesis: The type of event ('type' column: earthquake, quarry blast, etc.) is independent of the day of week
(DayOfWeek) on which it occurred.

In [ ]:
contingency = pd.crosstab(df['type'], df['DayOfWeek'])
chi2, p, dof, expected = chi2_contingency(contingency)
print('Chi-square statistic:', chi2)
print('Degrees of freedom:', dof)
print('p-value:', p)

Chi-square statistic: 92.16992465854213
Degrees of freedom: 36
p-value: 8.07286408381138e-07


**Statement**: The chi-square test result for this hypothesis yields a chi-square statistic of 92.17, degrees of freedom of 36, and a p-value of approximately $8.072 \times 10^{-07}$
 . Since this p-value is far below the conventional significance threshold of 0.05, there is very strong evidence to reject the null hypothesis. Therefore, the day of the week and event type are significantly dependent, indicating a meaningful association between these categorical variables in your data.

#### 5. Magnitude Mean Difference by Year Decade
Hypothesis: The mean earthquake magnitude (mag) differs significantly between records from two different decades (e.g., 2000s vs 2010s).

In [ ]:
df['decade'] = (df['Year'] // 10) * 10
dec1 = df[df['decade'] == 2000]['mag'].dropna()
dec2 = df[df['decade'] == 2010]['mag'].dropna()

ttest_result = ttest_ind(dec1, dec2, equal_var=False)
print('t-test statistic:', ttest_result.statistic)
print('p-value:', ttest_result.pvalue)

t-test statistic: 7.034358911416218
p-value: 2.0392846020245177e-12


**Statement**: The t-test result for this hypothesis yields a t-test statistic of 7.03 and a p-value of approximately $2.039 \times 10^{-12}$. Since this p-value is extremely small and far below the conventional significance threshold of 0.05, there is very strong evidence to reject the null hypothesis. Therefore, the mean earthquake magnitude differs significantly between the two compared decades in your dataset.

### Task:

- Suggest five additional hypothesis tests, based on the preprocessed earthquake dataset, that cover both numerical and categorical variables using t-tests and chi-square tests, to extend the depth of your analysis.

6. Depth Mean Difference by Review Status

Hypothesis:
The mean depth of earthquakes differs significantly between reviewed events and non-reviewed events.

In [ ]:
from scipy import stats
import numpy as np

alpha = 0.05

# make groups
g1 = df[df['status']=='reviewed']['depth'].dropna()
g2 = df[df['status']!='reviewed']['depth'].dropna()

# check variance equality
lev_stat, lev_p = stats.levene(g1, g2)
equal_var = (lev_p > alpha)

tstat, pval = stats.ttest_ind(g1, g2, equal_var=equal_var)
print("Levene p:", lev_p, "t:", tstat, "p:", pval)
if pval < alpha:
    print("Reject H0: mean depths differ")
else:
    print("Fail to reject H0: no evidence of difference")


Statement:
The t-test result for this hypothesis yields a test statistic and p-value.
If the p-value is less than the significance threshold of 0.05, there is sufficient evidence to reject the null hypothesis.
Thus, the mean depth of earthquakes differs significantly between reviewed and non-reviewed events.
However, if the p-value is greater than 0.05, then there is not enough evidence to reject the null hypothesis, meaning the mean depth does not differ significantly.

7. Magnitude Difference Between Shallow and Deep Earthquakes

Hypothesis:
The mean earthquake magnitude differs significantly between shallow earthquakes (depth < 70 km) and deep earthquakes (depth ≥ 70 km).

In [ ]:
alpha = 0.05
shallow = df[df['depth'] < 70]['mag'].dropna()
deep = df[df['depth'] >= 70]['mag'].dropna()

lev_stat, lev_p = stats.levene(shallow, deep)
equal_var = (lev_p > alpha)
tstat, pval = stats.ttest_ind(shallow, deep, equal_var=equal_var)
print("Levene p:", lev_p, "t:", tstat, "p:", pval)


Statement:
The t-test result for this hypothesis yields a test statistic and p-value.
If the p-value is below 0.05, there is strong evidence to reject the null hypothesis.
Thus, the mean magnitude is significantly different between shallow and deep earthquakes.
If the p-value is above 0.05, then there is insufficient evidence to conclude any meaningful difference in mean magnitude between the two groups.

8. Magnitude Category Consistency Across Seasons

Hypothesis:
There is a significant association between the season of occurrence and the magnitude category (low, medium, high, very high) of earthquakes.

In [ ]:
# define mag bins and seasons
bins = [0, 3.5, 5.0, 6.5, 10]   # adjust as wanted
labels = ['low','medium','high','very_high']
df['mag_bin'] = pd.cut(df['mag'], bins=bins, labels=labels, include_lowest=True)

def month_to_season(m):
    if m in [12,1,2]: return 'winter'
    if m in [3,4,5]: return 'spring'
    if m in [6,7,8]: return 'summer'
    return 'autumn'

df['season'] = df['month'].dropna().astype(int).apply(month_to_season)

cont = pd.crosstab(df['mag_bin'], df['season'])
chi2, p, dof, exp = stats.chi2_contingency(cont)
print("chi2:", chi2, "p:", p, "dof:", dof)
if p < 0.05:
    print("Reject H0: mag bin and season are associated")
else:
    print("Fail to reject H0")


Statement:
The chi-square test result for this hypothesis provides a chi-square statistic and p-value.
If the p-value is less than 0.05, we reject the null hypothesis.
Therefore, magnitude category and season are dependent variables, meaning earthquake magnitude levels vary across seasons.
If the p-value is greater than 0.05, then we fail to reject the null hypothesis, indicating no significant association between season and magnitude category.

9. Event Type Consistency Across Decades

Hypothesis:
The type of earthquake event (e.g., earthquake, quarry blast, etc.) is significantly associated with the decade in which the event occurred.

In [ ]:
cont = pd.crosstab(df['type'], df['decade'])
chi2, p, dof, exp = stats.chi2_contingency(cont)
print("chi2:", chi2, "p:", p)
if p < 0.05:
    print("Reject H0: type and decade are dependent")
else:
    print("Fail to reject H0")


Statement:
The chi-square test yields a chi-square statistic and p-value.
If the p-value is below 0.05, there is strong evidence to reject the null hypothesis.
Thus, event type and decade are dependent, indicating that the distribution of event types varies across decades.
If the p-value is greater than 0.05, then there is insufficient evidence to suggest any meaningful association between event type and decade.

10. Magnitude Mean Difference Between Northern and Southern Hemisphere

Hypothesis:
The mean earthquake magnitude differs significantly between events occurring in the Northern Hemisphere and those in the Southern Hemisphere.

In [ ]:
north = df[df['latitude'] > 0]['mag'].dropna()
south = df[df['latitude'] < 0]['mag'].dropna()

lev_stat, lev_p = stats.levene(north, south)
equal_var = (lev_p > 0.05)
tstat, pval = stats.ttest_ind(north, south, equal_var=equal_var)
print("Levene p:", lev_p, "t:", tstat, "p:", pval)


Statement:
The t-test result for this hypothesis provides a test statistic and p-value.
If the p-value is less than 0.05, the null hypothesis can be rejected.
Thus, the mean earthquake magnitude differs significantly between the Northern and Southern Hemispheres.
If the p-value is greater than 0.05, then there is not enough evidence to conclude a significant difference in mean magnitude between hemispheres.